packages/editor/src/components/document-outline/check.js

```diff
@@ -1,17 +1,11 @@
-/**
- * External dependencies
- */
-import { filter } from 'lodash';
-
 /**
  * WordPress dependencies
  */
 import { withSelect } from '@wordpress/data';
 import { store as blockEditorStore } from '@wordpress/block-editor';
 
 function DocumentOutlineCheck( { blocks, children } ) {
-	const headings = filter(
-		blocks,
+	const headings = blocks.filter(
```

Comment:
`blocks` will always be an array:

https://github.com/WordPress/gutenberg/blob/8d4958b59b4ebf20217ce36acd5fa42dbd782382/packages/block-editor/src/store/selectors.js#L174